domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -161,101 +192,310 @@ class TopicListController @Inject constructor(
-              story.chapterCount,
-              recentlyPlayerChapterSummary.name,
-              recentlyPlayerChapterSummary.explorationId
-            )
-            if (numberOfDaysPassed < ONE_WEEK_IN_DAYS) {
-              ongoingStoryListBuilder.addRecentStory(promotedStory)
-            } else {
-              ongoingStoryListBuilder.addOlderStory(promotedStory)
+              completedStoryTopicId
+            ).let {
+              Pair(it.first, it.second)
+              numberOfDaysPassed = it.second
+              it.first?.let { it1 -> recentlyPlayedPromotedStoryList.add(it1) }
             }
           }
-        } else if (lastCompletedChapterProgress != null &&
-          lastCompletedChapterProgress.explorationId != story.chapterList.last().explorationId
-        ) {
-          val lastChapterSummary: ChapterSummary? = story.chapterList.find { chapterSummary ->
-            lastCompletedChapterProgress.explorationId == chapterSummary.explorationId
-          }
-          val nextChapterIndex = story.chapterList.indexOf(lastChapterSummary) + 1
-          val nextChapterSummary: ChapterSummary? = story.chapterList[nextChapterIndex]
-          if (nextChapterSummary != null) {
-            val numberOfDaysPassed =
-              (Date().time - lastCompletedChapterProgress.lastPlayedTimestamp) / ONE_DAY_IN_MS
-            val promotedStory = createPromotedStory(
+          mostRecentCompletedChapterProgress != null &&
+            mostRecentCompletedChapterProgress.explorationId !=
+            story.chapterList.last().explorationId -> {
+            createOngoingStoryListBasedOnMostRecentlyCompleted(
               storyId,
+              story,
+              mostRecentCompletedChapterProgress,
+              completedChapterProgressList,
               topic,
-              completedChapterProgressList.size,
-              story.chapterCount,
-              nextChapterSummary.name,
-              nextChapterSummary.explorationId
-            )
-            if (numberOfDaysPassed < ONE_WEEK_IN_DAYS) {
-              ongoingStoryListBuilder.addRecentStory(promotedStory)
-            } else {
-              ongoingStoryListBuilder.addOlderStory(promotedStory)
+              completedStoryTopicId
+            ).let {
+              Pair(it.first, it.second)
+              numberOfDaysPassed = it.second
+              it.first?.let { it1 -> recentlyPlayedPromotedStoryList.add(it1) }
             }
           }
         }
       }
     }
-    if ((ongoingStoryListBuilder.olderStoryCount + ongoingStoryListBuilder.recentStoryCount) == 0) {
-      ongoingStoryListBuilder.addAllRecentStory(createRecommendedStoryList())
+    return Pair(recentlyPlayedPromotedStoryList, numberOfDaysPassed)
+  }
+
+  private fun checkIfStoryIsCompleted(
+    topicId: String,
+    mostRecentCompletedChapterProgress: ChapterProgress?,
+    story: StorySummary
+  ) {
+    if (mostRecentCompletedChapterProgress != null &&
```

Comment:
You can remove this null check by using this code:
```
    if (mostRecentCompletedChapterProgress?.explorationId == story.chapterList.last().explorationId) {
      completedStoryTopicId = topicId
    }
```

Reply:
Done.